govcd/adminorg.go

```diff
@@ -42,6 +42,74 @@ func (adminOrg *AdminOrg) CreateCatalog(name, description string) (AdminCatalog,
 	return CreateCatalog(adminOrg.client, adminOrg.AdminOrg.Link, name, description)
 }
 
+// CreateCatalogWithStorageProfile is like CreateCatalog, but allows to specify storage profile
+func (adminOrg *AdminOrg) CreateCatalogWithStorageProfile(name, description string, storageProfiles *types.CatalogStorageProfiles) (AdminCatalog, error) {
```

Comment:
Let's follow our guidelines and return referenced AdminCatalog

Reply:
Can do. The reason here was that it is "sister" function of `CreateCatalog`.

Reply:
Well, that "sister" is granny with older standard :)

Reply:
Ok, let them betray the family :)